packages/navitem/src/react/common.tsx

```diff
@@ -0,0 +1,47 @@
+import Halo from '@pluralsight/ps-design-system-halo'
+import { PropsOf, ValueOf, stylesFor } from '@pluralsight/ps-design-system-util'
+import { compose, css } from 'glamor'
+import React, { useContext } from 'react'
+
+import stylesheet from '../css'
+import { alignments } from '../vars'
+
+import Context from './context'
+
+const styles = {
+  bar: (props: {
+    alignment: ValueOf<typeof alignments>
+    selected: boolean
+  }) => {
+    const isHorz = (props.alignment = alignments.horizontal)
+    const isVert = (props.alignment = alignments.vertical)
+
+    return compose(
+      css(stylesheet['.psds-navitem__bar']),
+      isHorz && css(stylesheet['.psds-navitem__horz-bar']),
+      isVert && css(stylesheet['.psds-navitem__vert-bar']),
+      stylesFor('navitem__bar', props),
+      props.selected && stylesFor('navitem__bar--selected', props)
+    )
+  },
+  button: (props: { selected: boolean }) =>
+    compose(
+      css(stylesheet['.psds-navitem__button']),
+      props.selected && css(stylesheet['.psds-navitem__button--selected'])
+    )
+}
+
+export const Bar: React.FC<PropsOf<'div'>> = props => {
```

Comment:
I believe that using the `PropsOf` util (which is doing `JSX.IntrinsicElements[TagName]`) is more complete than `HTMLAttributes<Element>`. here is an example of the difference -> https://codesandbox.io/s/wonderful-night-c38uw?file=/src/Button.test.tsx

you'll see that when using `HTMLAttributes` - typescript can't completely type all button attrs - `type` is not typed

![Screen Shot 2020-11-09 at 12 32 32 PM](https://user-images.githubusercontent.com/226356/98587742-cc648080-2287-11eb-9b8a-44d54b1a9d93.png)

Reply:
Would the union type issue pop up here?

Reply:
it's not necessary for the Bar component since it's a `div` - just the first part of the review where i hit the typing so i'm calling it out. i'm trying to show that the difference between `HTMLAttributes` and `JSX.IntrinsicElements` is a important one and that `JSX.IntrinsicElements` provides a higher level of introspection. Since it's the more complete typing i'd venture we'd want to use it everywhere.

`AllHTMLAttributes` would include the `type` prop but accept any `string`. `JSX.IntrinsicElements` properly types the `type` attribute to only include values accepted on a `button` which provides greater introspection. `AllHTMLAttributes` also includes all possible attributes for **any** element - which seem way to wide and error prone.

Reply:
^ Thanks for your additional observations on `AllHTMLAttributes`.  The latter point, esp, I had missed "AllHTMLAttributes also includes all possible attributes for any element".  I agree: too wide and unuseful.